mkdocs/docs/HPC/setting_up_python_virtual_environments.md

```diff
@@ -0,0 +1,221 @@
+# Python Virtual Environments (venv's)
+
+## Introduction
+
+Python virtual environments (venv's) are a way to create isolated Python environments.
+You can install packages in a venv without affecting the system Python installation.
+Because a normal user cannot install packages globally, 
+venv's are a way to install packages without needing root access.
+This is useful when you need to use a package that is not available as a module on the HPC cluster.
+
+## Creating a venv
+
+A venv can be created with the following commands:
+
+```bash
+$ python -m venv myenv      # Create a new venv named myenv
+$ source myenv/bin/activate # Activate the venv
+```
+
+After activating the venv, you can install packages with `pip`:
+
+```bash
+$ pip install example_package1
+$ pip install example_package2
+```
+
+It is now possible to run Python scripts that use the installed packages in the venv. To deactivate the venv, run:
+
+```bash
+$ deactivate
+```
+
+!!! note
+    Always prefer to use modules if they are available. 
+    Modules are compiled and optimized for the HPC cluster, while packages installed with `pip` are not.
+    To check if a package is available as a module, you can use the following command:
+
+    ```bash
+    $ module av package_name
+    ```
+
+## Combining venv's with modules
+
+You can combine packages installed in a venv with modules. The following script uses 
+pytorch (which is available as a module) and Poutyne (which is not available as a module):
+
+```python title="pytorch_poutine.py"
+import torch
+import poutyne
+
+...
+```
+
+We load a PyTorch package as a module and install Poutyne in a venv:
+
+```bash
+$ ml PyTorch/2.1.2-foss-2023a
+$ python -m venv myenv
+$ source myenv/bin/activate
+$ pip install Poutyne
+```
+
+While in the virtual environment, we can run the script without any issues:
+
+```bash
+$ python pytorch_poutine.py
+```
+
+
+## Creating a virtual environment on a specific cluster
+
+To create a virtual environment for a specific cluster, you need to start an interactive shell on that cluster.
+Let's say you want to create a virtual environment on the `shinx` cluster.
+
+```bash
+$ module swap cluster/shinx
+$ qsub -I
+```
```

Comment:
This problem occurs because software install for cluster `shinx` (which is running RHEL9) is not compatible with the `gligar07` and `gligar08` login nodes (which run RHEL8)

Reply:
@lbarraga Please use a different cluster for now, `donphan` for example.